File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.rb
# @param {Integer[][]} points
# @return {Integer}
def find_min_arrow_shots(points)
    points.sort!{ |a, b| a[0] <=> b[0] }
    # points.sort_by! { |a, b| a }
    
    arrow = 1
    end_value = points[0][1]

    points.each do |point|
        if point[0] > end_value
            arrow += 1
            end_value = point[1]
        else
            end_value = [end_value, point[1]].min
        end
    end
    
    return arrow
end

File: 452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end(), [](const auto &a, const auto &b){ return a[0] < b[0]; });

        int arrow = 1;
        int end = points[0][1];
        for(auto x:points){
            if(x[0] > end){
                end = x[1];
                arrow++;
            }else{
                end = min(x[1], end);
            }
        }
        return arrow;
    }
};